### DISCUSSION TITLE
How to disable autoplay at instagram.com?

### DISCUSSION THREAD

**Discussion by Vanerio** [uBlockOrigin/uBlock-issues#3098] [Q&A] [answered]
Hi!
I have tried these filters to make use of Permissions Policy:
`||instagram.com^$permissions=autoplay=()`
`*$permissions=autoplay=()`
But they do not work.
Am I missing something? Should I do something different?
Thank you very much!

**Comment by iam-py-test** [comment]
The `autoplay` permissions policy is not supported in Firefox: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Permissions-Policy/autoplay#browser_compatibility
Thanks

**Comment by Vanerio** [comment]
I am using Brave, not Firefox
Could this be a problem?
Thank you!

**Reply by iam-py-test** [reply]
Ok, sorry.
Would you mind:
- click the uBlock Origin icon *while on Instagram* (with the filter added)
- click the chat icon
- scroll down to _Troubleshooting information_
- copy that here

Thanks

**Reply by Vanerio** [reply]
Here you are:

> uBlock Origin: 1.55.0
> Chromium: 121
> filterset (summary):
>  network: 292515
>  cosmetic: 124318
>  scriptlet: 25913
>  html: 0
> listset (total-discarded, last-updated):
>  added:
>   https://raw.githubusercontent.com/DandelionSprout/adfilt/master/LegitimateURLShortener.txt: 2555-1, 38m
>   https://raw.githubusercontent.com/Spam404/lists/master/adblock-list.txt: 7253-0, 4h.21m
>   https://secure.fanboy.co.nz/fanboy-annoyance_ubo.txt: 95716-16366, 5d.22h.24m
>   adguard-spyware: 89699-172, 4d.18h.14m
>   adguard-spyware-url: 1424-156, 4d.18h.15m
>   curben-phishing: 86522-20, 37m
>   dpollock-0: 11632-200, 1d.22h.42m
>   fanboy-cookiemonster: 49426-49299, 21h.23m
>   fanboy-social: 16115-16107, 21h.23m
>   fanboy-thirdparty_social: 68-15, 5d.22h.24m
>   ublock-cookies-easylist: 912-2, 21h.24m
>  default:
>   user-filters: 520-0, never
>   easylist: 79963-332, 3h.16m Δ
>   easyprivacy: 43571-26380, 3h.16m Δ
>   plowe-0: 3786-1613, 1d.22h.41m
>   spa-0: 3838-21, 38m
>   spa-1: 5730-224, 4d.18h.15m
>   ublock-badware: 7796-148, 3h.16m Δ
>   ublock-filters: 36821-363, 3h.16m Δ
>   ublock-privacy: 992-425, 3h.16m Δ
>   ublock-quick-fixes: 143-9, 3h.16m Δ
>   ublock-unbreak: 2198-83, 3h.16m Δ
>   urlhaus-1: 8160-0, 38m
> filterset (user): [array of 510 redacted]
> switchRuleset:
>  added: [array of 157 redacted]
> hostRuleset:
>  added: [array of 290 redacted]
> userSettings:
>  advancedUserEnabled: true
>  showIconBadge: false
>  suspendUntilListsAreLoaded: true
> hiddenSettings:
>  trustedListPrefixes: ublock- user-
> supportStats:
>  allReadyAfter: 369 ms (selfie)
>  maxAssetCacheWait: 259 ms
>  cacheBackend: browser.storage.local
> popupPanel:
>  blocked: 4
>  no-remote-fonts: true
>  network:
>   facebook.com: 4
>  extended:
>   ##._aagw
>   ##._a9-7
>   ##article:style(margin-bottom: 69px !important;)
>   ##.x1porb0y.x6osk4m.x17snn68.x14k21rp.x10o80wk.xh8yej3.x1gryazu:…
>   ##+js(trusted-set-cookie, ig_did, 0C826C21-17C3-444A-ABB7-EBABD3…
>   ##+js(addEventListener-defuser, dblclick)

**Reply by uBlock-user** [reply, marked as answer]
Brave has built in autoplay blocker. Configure that.

brave://settings/content/autoplay?search=autoplay

**Reply by Vanerio** [reply]
Thank you, your suggestion works!. However I would still like to know how to do it with uBlock and of course know if I am doing something wrong with the filters.

**Comment by Vanerio** [comment]
Still not working in version 1.56.0
Any help would be greatly appreciated

**Reply by uBlock-user** [reply]
What do you mean still not working ? 

You said my suggestion worked as per https://github.com/uBlockOrigin/uBlock-issues/discussions/3098#discussioncomment-8272183

**Reply by Vanerio** [reply]
Thank you for trying to help. Yes, using Brave conf stops autoplay.
But I want to do it with a uBlock Origin filter. And it is not working at the moment, at least with the examples I wrote.

**Reply by Vanerio** [reply]
I don´t know why you marked as answered, even if my question has not been answered. Your suggestion is just a 'workaround' out of uBo

**Reply by uBlock-user** [reply]
If `$permissions` doesn't work then, currently nothing else will.

**Reply by Vanerio** [reply]
Do you mean that $permissions is not working in uBo at the moment?
I might have written the filters wrong though, as I am not an expert.
Thank you

**Reply by Vanerio** [reply]
The Brave native solution sometimes does not work...If you play a video the block-autoplay setting stops working sometimes.
Still wondering what should I do to block it with uBo!
Any help will be greatly appreciated

**Reply by krystian3w** [reply]
It remains to analyze a ton of JavaScript on this site.

Once you've noticed how the site adds autoplay (instead simple autoplay attribute visible in `view-source:`), you can consider stop it some of the JS code at Instagram after to match these to uBo's arsenal of scriptlets.

In uBo with Firefox the `replace` modifier could work, by removing the autoplay attribute in `view-source` when fail **:remove-attr(autoplay)** and **ra.js** with `asap stay` tribe (may need killing the dynamic loading of subpages).

RU AdList clean up some autoplay in players: 

https://pastebin.com/raw/NSuehATZ (but not Instagram yet in main file)

**Comment by r00t-** [comment]
there is a standlone plugin for that functionality here (i am NOT the author of it):
https://github.com/ReeceHarding/instagram-autoplay-neutralizer
so the analysis has been done there already.
one could try to port it into a uBlock configuration.

also, the EU is working on disabling autoplay by **legislative** means: https://arstechnica.com/tech-policy/2026/07/disable-auto-play-and-infinite-scroll-or-risk-massive-fines-eu-tells-meta/
